src/ControlSystem/Systems/Size.hpp

```diff
@@ -0,0 +1,194 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <optional>
+#include <string>
+
+#include "ApparentHorizons/StrahlkorperInDifferentFrame.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "ControlSystem/Component.hpp"
+#include "ControlSystem/ControlErrors/Size.hpp"
+#include "ControlSystem/Measurements/CharSpeed.hpp"
+#include "ControlSystem/Protocols/ControlError.hpp"
+#include "ControlSystem/Protocols/ControlSystem.hpp"
+#include "ControlSystem/Protocols/Measurement.hpp"
+#include "ControlSystem/Tags/QueueTags.hpp"
+#include "ControlSystem/Tags/SystemTags.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/LinkedMessageQueue.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Structure/ObjectLabel.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Tags.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Printf.hpp"
+#include "ParallelAlgorithms/Actions/UpdateMessageQueue.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Frame {
+struct Grid;
+struct Distorted;
+}  // namespace Frame
+namespace control_system {
+template <typename ControlSystem>
+struct UpdateControlSystem;
+}
+/// \endcond
+
+namespace control_system::Systems {
+/*!
+ * \brief Controls the \f$l=0\f$ component of the \link
+ * domain::CoordinateMaps::TimeDependent::Shape Shape \endlink map
+ *
+ * Requirements:
+ * - This control system requires that there be at least one object in the
+ *   simulation
+ * - This object must be a black hole (have an excision)
+ * - Currently this control system can only be used with the \link
+ *   control_system::measurements::CharSpeed CharSpeed \endlink
+ * measurement
+ * - Currently this control system can only be used with the \link
+ *   control_system::ControlErrors::Size Size \endlink control error
+ */
+template <::domain::ObjectLabel Horizon, size_t DerivOrder>
```

Comment:
How would I choose the deriv order? Maybe add docs here or to the protocol (can be another PR)

Reply:
By trial and error basically. Or know that deriv order 2 was found to work well in spec for most control systems (except a notable couple). Either way, I'll do this in another PR. I know I need to add an entire explanation on how to control system communication works anyways.